.github/workflows/auto-approve.yml

```diff
@@ -1,15 +1,55 @@
 # Approve PRs with "pr/auto-approve". mergify takes care of the actual merge.
 
 name: auto-approve
-on: pull_request
+on:
+  pull_request:
+    types:
+      - labeled           # Run when labels are updated
+      - opened            # Run when a PR is initially submitted
+      - ready_for_review  # Run when a Draft PR is upgraded PR
+      - reopened          # Run when a PR is re-opened
+      - synchronize       # Run when a PR head branch is updated
 
 jobs:
   auto-approve:
-    if: contains(github.event.pull_request.labels.*.name, 'pr/auto-approve')
+    if: contains(github.event.pull_request.labels.*.name, 'pr/auto-approve') || github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]'
+
     permissions:
       pull-requests: write
     runs-on: ubuntu-latest
+
     steps:
-    - uses: hmarr/auto-approve-action@v2.1.0
-      with:
-        github-token: "${{ secrets.GITHUB_TOKEN }}"
\ No newline at end of file
+      # Check whether the PR needs an automated approval or not. We will only add an automated
+      # approval if the PR currently has no reviews on it. This way, the automated workflow will not
+      # do anything if a manual review has already been done for the PR (implicit opt-out).
```

Comment:
A review might be created after the PR is created but before this workflow runs. I'd agree this would be a relatively uncommon occurrence.

The auto-approval could however get discarded, and I don't know what this manifests like...

What I wanted to achieve is avoiding a pile of approvals to be created on PRs as they go through cycles of "update branch", because this makes the PR history difficult to navigate.

Reply:
> What I wanted to achieve is avoiding a pile of approvals to be created on PRs as they go through cycles of "update branch", because this makes the PR history difficult to navigate.

In that case I suggest either checking if the PR is already approved, or removing the "synchronize" trigger. Also I think that in the current state, if the approval is discarded, it won't get auto approved again since a comment has already been made by the bot in the previous approval...

I'm inclined to either change the condition or leave it be.

Reply:
> Updated to ignore discarded reviews

Not seeing it...sure you committed it?

Reply:
Eh! My push had failed :)